node/src/main/scala/coop/rchain/node/effects/RchainEvents.scala

```diff
@@ -19,7 +19,7 @@ trait RchainEvents[F[_], G[_]] extends EventPublisher[F] with EventConsumer[G]
 object RchainEvents {
   def readerTInstance[F[_]: Sync: Concurrent, E]: ReaderT[F, E, RchainEvents[ReaderT[F, E, ?], F]] =
     for {
-      q <- Queue.in[ReaderT[F, E, ?]].circularBuffer[F, RChainEvent](1000)
+      q <- Queue.in[ReaderT[F, E, ?]].circularBuffer[F, RChainEvent](1)
```

Comment:
what happens when there's two produces before a consume happens?

Reply:
Apologies, but the extended description doesn't make it clear to me:
will the buffer we've ended up using block when it's full? Or will it drop messages? (I think we're using a buffer of size 1 in the end?)

Reply:
no worries, IMO explaining oneself is a good habit.

the circular buffer will overwrite if it is not drained. this is fine - if there is no one draining we don't want to block.
on the other hand, the topic is set up to drain the queue so the size of it does not matter.

Reply:
@sebtomba have you read my response? if yes - which part is not clear?

Reply:
@dzajkowski Sorry, GH didn't show me your response.
The question is not if it's drained but if it's drained fast enough at all times. A buffer serves the purpose to mitigate event production peaks. For that reason, I would recommend making the buffer size > 1 (f.e. 100).